src/features/identityCheck/pages/profile/StatusFlatList.tsx

```diff
@@ -0,0 +1,162 @@
+import colorAlpha from 'color-alpha'
+import React, { useRef, useState } from 'react'
+import { Control, Controller, UseFormHandleSubmit } from 'react-hook-form'
+import { View, FlatList, ViewStyle, LayoutChangeEvent, ListRenderItem } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import styled from 'styled-components/native'
+
+import { ActivityIdEnum, ActivityResponseModel } from 'api/gen'
+import { useActivityTypes } from 'features/identityCheck/api/useActivityTypes'
+import { useOnViewableItemsChanged } from 'features/subscription/helpers/useOnViewableItemsChanged'
+import { AccessibilityRole } from 'libs/accessibilityRole/accessibilityRole'
+import { AnimatedViewRefType, createAnimatableComponent } from 'libs/react-native-animatable'
+import { theme } from 'theme'
+import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
+import { Li } from 'ui/components/Li'
+import { RadioSelector } from 'ui/components/radioSelector/RadioSelector'
+import { Spinner } from 'ui/components/Spinner'
+import { getSpacing, Spacer, Typo } from 'ui/theme'
+
+const GRADIENT_HEIGHT = getSpacing(30)
+const VIEWABILITY_CONFIG = { itemVisiblePercentThreshold: 100 }
+
+export type StatusForm = {
+  selectedStatus: ActivityIdEnum | null
+}
+
+interface Props {
+  handleSubmit: UseFormHandleSubmit<StatusForm>
+  isLoading: boolean
+  selectedStatus: ActivityIdEnum | null
+  submitStatus: (formValues: StatusForm) => Promise<void>
+  titleID: string
+  control: Control<StatusForm>
+  headerHeight: number
+}
+
+export function StatusFlatList({
+  handleSubmit,
+  isLoading,
+  selectedStatus,
+  submitStatus,
+  titleID,
+  control,
+  headerHeight,
+}: Props) {
+  const { activities } = useActivityTypes()
+
+  const gradientRef = useRef<AnimatedViewRefType>(null)
+
+  const { onViewableItemsChanged } = useOnViewableItemsChanged(gradientRef, activities ?? [])
+
+  const [bottomViewHeight, setBottomViewHeight] = useState(0)
+
+  function onBottomViewLayout(event: LayoutChangeEvent) {
+    const { height } = event.nativeEvent.layout
+    setBottomViewHeight(height)
+  }
+
+  const renderItem: ListRenderItem<ActivityResponseModel> = ({ item }) => (
+    <View accessibilityRole={AccessibilityRole.RADIOGROUP} accessibilityLabelledBy={titleID}>
+      <Li key={item.label}>
+        <Controller
+          control={control}
+          name="selectedStatus"
+          render={({ field: { value, onChange } }) => (
+            <RadioSelector
+              checked={item.id === value}
+              label={item.label}
+              description={item.description}
+              onPress={() => onChange(item.id)}
+            />
+          )}
+        />
+        <Spacer.Column numberOfSpaces={3} />
+      </Li>
+    </View>
+  )
+
+  return (
+    <React.Fragment>
+      {activities ? (
+        <FlatListContainer>
+          <FlatList
+            scrollIndicatorInsets={{ right: 1 }} // Corrects scrollbar in the middle
```

Comment:
Style externe ?

Reply:
Le problème quand je crée un styled(FlatList) c'est que le typage "se perd" et je me retrouve avec des erreurs sur renderItem et keyExtractor. Je suis prenneur de solutions autre que ce qui existe dans notre code avec du forçage de type ("as ..."). J'ai vu un example de styled(FlatList) dans src/features/home/components/modules/VenueListModule.tsx et ils ont fait const StyledFlatList = styled(FlatList as typeof FlatList<VenueHit>) que je trouve pas ouf!

Reply:
Au delà du typage il y a le rerendu...
Là en faisant un style interne, c'est un objet, dont la réf change à chaque rendu.

Reply:
Donc il vaut mieux que je fasse un const StyledFlatList = styled(FlatList as typeof FlatList) pour eviter le rerendu?

Reply:
Il est possible que ce soit une sur-optimisation inutile.
Reassure devrait nous aider sur ce point.
Eventuellement DevTools/Profiler.

Dans l'immédiat je ne bloquerai pas la validation de la PR juste pour ça car il faudrait juste tester les perfs à cet endroit, ce que nous ne savons pas encore bien faire. Mais j'y travaille :)